porch/pkg/engine/init.go

```diff
@@ -57,13 +57,13 @@ func (m *initPackageMutation) Apply(ctx context.Context, resources repository.Pa
 		Site:     m.task.Init.Site,
 	})
 	if err != nil {
-		return repository.PackageResources{}, nil, fmt.Errorf("failed to initialize pkg %q: %w", m.name, err)
+		return repository.PackageResources{}, nil, nil, fmt.Errorf("failed to initialize pkg %q: %w", m.name, err)
 	}
 
 	result, err := readResources(fs)
 	if err != nil {
-		return repository.PackageResources{}, nil, err
+		return repository.PackageResources{}, nil, nil, err
 	}
 
-	return result, m.task, nil
+	return result, nil, m.task, nil
```

Comment:
Nit: four retvals is likely where I start to think about a struct

Reply:
along these lines, perhaps there could be a struct that contains both `Task` and its `TaskResult`?

Reply:
Now returning `TaskResult` that contains `*Task`. Thanks.